recipes/fastjet/meta.yaml

```diff
@@ -0,0 +1,72 @@
+{% set name = "fastjet" %}
+{% set version = "3.4.0.6" %}
+
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/fastjet-{{ version }}.tar.gz
+  sha256: 86dea05ce7a030db9fe14e5bf1c0bc46351b6e439f9b9ef6b14a55ceefcc7b7a
+  patches:
+    - 0001-Don-t-overwrite-external-CXXFLAGS-and-LDFLAGS.patch
+
+build:
+  number: 0
+  skip: true  # [not linux]
+  script: {{ PYTHON }} -m pip install . -vv
+  missing_dso_whitelist:
+  # https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#whitelisting-shared-libraries
+  # All are under lib/python3.*/site-packages/fastjet/_fastjet_core/lib
+    - $RPATH/libfastjet.so.0
+    - $RPATH/libfastjettools.so.0
+    - $RPATH/libsiscone.so.0
+    - $RPATH/libsiscone_spherical.so.0
+    - $RPATH/libfastjetplugins.so.0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - make
+    - libtool
+    - automake
+    - autoconf
+    - swig
+    - patch
+    - gmp  # In 'build' over 'host' as avoids 'ld: cannot find -lgmp: No such file or directory' error
+  host:
+    - pip
+    - pybind11
+    - python
+    - cgal-cpp
+    - gmp
```

Comment:
@lgray I might be wrong, @xhochy correct me if so, but I think if the build only works if we have a package listed in both `build` and `host` then it means that the `host` package isn't getting correctly picked up and things will be broken for the reasons that @xhochy pointed out in https://github.com/conda-forge/staged-recipes/pull/21052/files#r1494036235.

I truly don't know though.

Reply:
Ah I think I see. I find the documentation opaque so I have resorted to cargo-cultism and taking further instructions at face value. Thus, I have added gmp to host. I have a feeling the way around it is to use cross-compiling explicitly in the build stage. However, I also do not know.

Reply:
Yeah, I was thinking also maybe the problem is that we haven't told the linker where to find `libgmp` yet and so we just need to patch in an `-L ${PREFIX}/lib` but nope as looking at my local build logs the build does this automatically for you

```console
$ grep --max-count 1 "LDFLAGS=" build-success.log 
+LDFLAGS=-Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib
```

so still confused.

Reply:
Hm, also looking at the error

```
  /home/conda/staged-recipes/build_artifacts/fastjet_*/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/12.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lgmp: No such file or directory
  collect2: error: ld returned 1 exit status
```

If I locally remove `gmp` from `build` and put it in `host` and then rebuild, if we look at where this `ld` is

```console
$ readlink -f build_artifacts/fastjet_*/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/12.3.0/../../../../x86_64-conda-linux-gnu/
/home/feickert/Code/GitHub/forks/staged-recipes/build_artifacts/fastjet_1708464123785/_build_env/x86_64-conda-linux-gnu
```

we notice that the this `ld` linker lives under the `_build_env/` build environment directory and not the `_h_env*/` host environment directory.

So if we look at what is under this linker's `lib/` we see that there's no `libgmp` as we moved it to `host` in the `meta.yml`.

```console
$ ls build_artifacts/fastjet_*/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/12.3.0/../../../../x86_64-conda-linux-gnu/lib/
ldscripts          libasan.so.8.0.0    libgcc_s.so       libgfortran.so.5.0.0  libitm.so.1      liblsan.so.0      libquadmath.so.0.0.0  libstdc++.so.6.0.30  libtsan.so.2.0.0
libasan_preinit.o  libatomic.so        libgcc_s.so.1     libgomp.so            libitm.so.1.0.0  liblsan.so.0.0.0  libsanitizer.spec     libtsan_preinit.o    libubsan.so
libasan.so         libatomic.so.1      libgfortran.so    libgomp.spec          libitm.spec      libquadmath.so    libstdc++.so          libtsan.so           libubsan.so.1
libasan.so.8       libatomic.so.1.2.0  libgfortran.so.5  libitm.so             liblsan.so       libquadmath.so.0  libstdc++.so.6        libtsan.so.2         libubsan.so.1.0.0
```

This also explains why it has been requiring `libgmp` to be in the `build` requirements, as the linker used is the `build` environment linker.

So we've been focusing on the wrong question when it comes to what `LDFLAGS` we need and should instead be figuring out how to get the correct environment's linker to be used.

And as a check, the linker does exist in both the `build` and `host` environments:

```console
$ find build_artifacts/fastjet_*/_build_env/ -type f -iname "x86_64-conda-linux-gnu-ld"
build_artifacts/fastjet_1708464123785/_build_env/bin/x86_64-conda-linux-gnu-ld
$ find build_artifacts/fastjet_*/_h_env*/ -type f -iname "x86_64-conda-linux-gnu-ld"
build_artifacts/fastjet_1708464123785/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/bin/x86_64-conda-linux-gnu-ld
```